src/MongoDB.Driver.Core/Core/Misc/ServerVersion.cs

```diff
@@ -0,0 +1,463 @@
+            if (result != 0)
+            {
+                return result;
+            }
+
+            if (_releaseCandidate != null || other._releaseCandidate != null)
+            {
+                if (_releaseCandidate == null)
+                {
+                    return 1;
+                }
+                if (other._releaseCandidate == null)
+                {
+                    return -1;
+                }
+
+                result = _releaseCandidate.Value.CompareTo(other._releaseCandidate.Value);
+                if (result != 0)
+                {
+                    return result;
+                }
+            }
+
+            if (_internalBuild != null || other._internalBuild != null)
+            {
+                if (_internalBuild == null)
+                {
+                    return -1;
+                }
+                if (other._internalBuild == null)
+                {
+                    return 1;
+                }
+
+                result = _internalBuild.Value.CompareTo(other._internalBuild.Value);
+                if (result != 0)
+                {
+                    return result;
+                }
+            }
+
+            if (_commitHash == null && other._commitHash == null)
+            {
+                return 0;
+            }
+            if (_commitHash == null)
+            {
+                return -1;
+            }
+            if (other._commitHash == null)
+            {
+                return 1;
+            }
+
+            return 0;
+        }
+
+        /// <inheritdoc/>
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as ServerVersion);
+        }
+
+        /// <inheritdoc/>
+        public bool Equals(ServerVersion other)
+        {
+            return CompareTo(other) == 0;
+        }
+
+        /// <inheritdoc/>
+        public override int GetHashCode()
+        {
+            return ToString().GetHashCode();
+        }
+
+        /// <inheritdoc/>
+        public override string ToString()
+        {
+            var sb = new StringBuilder($"{_major}.{_minor}.{_patch}");
+            if (_releaseCandidate != null)
+            {
+                sb.Append($"-rc{_releaseCandidate}");
+            }
+            if (_internalBuild != null)
+            {
+                sb.Append($"-{_internalBuild}");
+            }
+            if (_commitHash != null)
+            {
+                sb.Append($"-g{_commitHash}");
+            }
+
+            return sb.ToString();
+        }
+
+        // static methods
+        /// <summary>
+        /// Parses a string representation of a server version.
+        /// </summary>
+        /// <param name="value">The string value to parse.</param>
+        /// <returns>A server version.</returns>
+        /// <exception cref="FormatException">value</exception>
+        public static ServerVersion Parse(string value)
+        {
+            ServerVersion result;
```

Comment:
nit: move result into `out var ..`

Reply:
Yes but... if we want minimal diffs between SemanticVersion and ServerVersion we should refrain from making minor changes like this.

Reply:
Since we made the change in SemanticVersion.cs make it here also so there won't be an unnecessary diff.